src/App.tsx

```diff
@@ -67,19 +71,28 @@
                   </div>
                 )}
 
-                {author && loaded && !hasError && posts.length === 0 && (
-                  <div className="notification is-warning" data-cy="NoPostsYet">
-                    No posts yet
-                  </div>
-                )}
+                {selectedUser &&
+                  postsLoaded &&
+                  !postsError &&
+                  posts.length === 0 && (
+                    <div
+                      className="notification is-warning"
+                      data-cy="NoPostsYet"
+                    >
+                      No posts yet
+                    </div>
+                  )}
```

Comment:
The condition for displaying a 'No posts yet' message is correct. Ensure that the logic for determining `posts.length === 0` is accurate and that posts are being fetched and stored correctly.